json-rpc/types/src/views.rs

```diff
@@ -47,28 +47,51 @@ impl AmountView {
 }
 
 #[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
-pub enum AccountRoleView {
-    #[serde(rename = "unknown")]
+pub struct ChildVASP {
+    pub parent_vasp_address: BytesView,
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
+pub struct ParentVASP {
+    pub human_name: String,
+    pub base_url: String,
+    pub expiration_time: u64,
+    pub compliance_key: BytesView,
+    pub num_children: u64,
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
+pub struct DesignatedDealer {
+    pub human_name: String,
+    pub base_url: String,
+    pub expiration_time: u64,
+    pub compliance_key: BytesView,
+    pub preburn_balances: Vec<AmountView>,
+    pub received_mint_events_key: BytesView,
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum AccountRoleViewV1 {
     Unknown,
-    #[serde(rename = "child_vasp")]
-    ChildVASP { parent_vasp_address: BytesView },
-    #[serde(rename = "parent_vasp")]
-    ParentVASP {
-        human_name: String,
-        base_url: String,
-        expiration_time: u64,
-        compliance_key: BytesView,
-        num_children: u64,
-    },
-    #[serde(rename = "designated_dealer")]
-    DesignatedDealer {
-        human_name: String,
-        base_url: String,
-        expiration_time: u64,
-        compliance_key: BytesView,
-        preburn_balances: Vec<AmountView>,
-        received_mint_events_key: BytesView,
-    },
+    ChildVasp(ChildVASP),
+    ParentVasp(ParentVASP),
+    DesignatedDealer(DesignatedDealer),
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
+#[serde(rename_all = "snake_case", tag = "type")]
```

Comment:
instead of relying on the automatic tokenization here , I think we should follow convention in the rest of the code, manually tag all enum